src/pps/pps.proto

```diff
@@ -339,95 +323,88 @@ enum PipelineState {
+  // pipeline in a given state and the state of the most recently created job,
+  // respectively.
   map<int32, int32> job_counts = 7;
-  string auth_token = 8;
-  JobState last_job_state = 9;
+  JobState last_job_state = 8;
 
   // parallelism tracks the literal number of workers that this pipeline should
   // run.
-  uint64 parallelism = 10;
+  uint64 parallelism = 9;
 
   // The pipeline type is stored here so that we can internally know the type of
-  // the pipeline without loading the spec from PFS - this field is not in the
-  // full PipelineInfo as that would be redundant.
+  // the pipeline without loading the spec from PFS.
   enum PipelineType {
     PIPELINE_TYPE_TRANSFORM = 0;
     PIPELINE_TYPE_SPOUT = 1;
     PIPELINE_TYPE_SERVICE = 2;
   }
-  PipelineType type = 11;
-}
-
-message PipelineInfo {
-  Pipeline pipeline = 1;
-  uint64 version = 2;
-  Transform transform = 3;
-  // tf_job encodes a Kubeflow TFJob spec. Pachyderm uses this to create TFJobs
-  // when running in a kubernetes cluster on which kubeflow has been installed.
-  // Exactly one of 'tf_job' and 'transform' should be set
-  TFJob tf_job = 4 [(gogoproto.customname) = "TFJob"];
-  ParallelismSpec parallelism_spec = 5;
-  Egress egress = 6;
-  google.protobuf.Timestamp created_at = 7;
-
-  // state indicates the current state of the pipeline. This is not stored in
-  // PFS along with the rest of this data structure--PPS.InspectPipeline fills
-  // it in
-  PipelineState state = 8;
-  // same for stopped field
-  bool stopped = 9;
-  string recent_error = 10;
-
-  int64 workers_requested = 11;
-  int64 workers_available = 12;
-
-  // job_counts and last_job_state indicates the number of jobs within this
-  // pipeline in a given state and the state of the most recently created job,
-  // respectively. This is not stored in PFS along with the rest of this data
-  // structure--PPS.InspectPipeline fills it in from the StoredPipelineInfo.
-  map<int32, int32> job_counts = 13;
-  JobState last_job_state = 14;
-
-  string output_branch = 15;
-  ResourceSpec resource_requests = 16;
-  ResourceSpec resource_limits = 17;
-  ResourceSpec sidecar_resource_limits = 18;
-  Input input = 19;
-  string description = 20;
-  string cache_size = 21;
-  string salt = 23;
-
-  // reason includes any error messages associated with a failed pipeline
-  string reason = 24;
-  int64 max_queue_size = 25;
-  Service service = 26;
-  Spout spout = 27;
-  DatumSetSpec datum_set_spec = 28;
-  google.protobuf.Duration datum_timeout = 29;
-  google.protobuf.Duration job_timeout = 30;
-  bool standby = 31;
-  int64 datum_tries = 32;
-  SchedulingSpec scheduling_spec = 33;
-  string pod_spec = 34;
-  string pod_patch = 35;
-  bool s3_out = 36;
-  Metadata metadata = 37;
-  string reprocess_spec = 38;
-  // Set at call time, not stored in etcd
-  int64 unclaimed_tasks = 39;
-  string worker_rc = 40;
-  bool autoscaling = 41;
+  PipelineType type = 10;
+
+  string auth_token = 11;
```

Comment:
`AuthToken` is a little special in that it's stored in the database, but we _don't_ want to return it to users.  I added a test that `ListPipeline` and `InspectPipeline` don't pass it through, but it would be nice if we could have something more bulletproof.